Tinyxml2Serializer.cpp

```diff
@@ -64,67 +64,63 @@ void CTinyxml2Serializer::AddXMLElementFloat(tinyxml2::XMLDocument* oXML, const
-
-    //xml.AddElem("Position");
-    //xml.AddAttrib("X", std::to_string(position.x).c_str());
-    //xml.AddAttrib("Y", std::to_string(position.y).c_str());
-    //xml.AddAttrib("Z", std::to_string(position.z).c_str());
-
-    //xml.AddElem("Rotation");
-    //xml.AddAttrib("X", std::to_string(rotation.x).c_str());
-    //xml.AddAttrib("Y", std::to_string(rotation.y).c_str());
-    //xml.AddAttrib("Z", std::to_string(rotation.z).c_str());
-    //xml.AddAttrib("W", std::to_string(rotation.w).c_str());
-
-    //xml.OutOfElem();
+    tinyxml2::XMLElement* transformElem = oXML.NewElement(name.c_str());
+    parentElem->InsertEndChild(transformElem);
+
+    tinyxml2::XMLElement* positionElem = oXML.NewElement("Position");
+    positionElem->SetAttribute("X", std::to_string(position.x).c_str());
+    positionElem->SetAttribute("Y", std::to_string(position.y).c_str());
+    positionElem->SetAttribute("Z", std::to_string(position.z).c_str());
+    transformElem->InsertEndChild(positionElem);
+
+    tinyxml2::XMLElement* rotationElem = oXML.NewElement("Rotation");
+    rotationElem->SetAttribute("X", std::to_string(rotation.x).c_str());
+    rotationElem->SetAttribute("Y", std::to_string(rotation.y).c_str());
+    rotationElem->SetAttribute("Z", std::to_string(rotation.z).c_str());
+    rotationElem->SetAttribute("W", std::to_string(rotation.w).c_str());
+    transformElem->InsertEndChild(rotationElem);
 }
 
-void CTinyxml2Serializer::AddXMLElementDOF(tinyxml2::XMLDocument& xml, const std::string& name, SDegreeOfFreedom degreeOfFreedoms)
+void CTinyxml2Serializer::AddXMLElementDOF(tinyxml2::XMLDocument& oXML, tinyxml2::XMLElement* parentElem, const std::string& name, const SDegreeOfFreedom& degreeOfFreedoms)
 {
-    //xml.AddElem(name.c_str());
-    //if (!std::isnan(degreeOfFreedoms.lowerBound) && !std::isnan(degreeOfFreedoms.upperBound))
-    //{
-    //    if (mnMajorVersion > 1 || mnMinorVersion > 21)
-    //    {
-    //        xml.IntoElem();
-    //        xml.AddElem("Constraint");
-    //    }
-    //    xml.AddAttrib("LowerBound", std::to_string(degreeOfFreedoms.lowerBound).c_str());
-    //    xml.AddAttrib("UpperBound", std::to_string(degreeOfFreedoms.upperBound).c_str());
-    //}
+    tinyxml2::XMLElement* dofElem = oXML.NewElement(name.c_str());
+    parentElem->InsertEndChild(dofElem);
 
-    //if (std::isnan(degreeOfFreedoms.lowerBound) || std::isnan(degreeOfFreedoms.upperBound) || (mnMajorVersion == 1 && mnMinorVersion < 22))
-    //{
-    //    xml.IntoElem();
-    //}
+    if (!std::isnan(degreeOfFreedoms.lowerBound) && !std::isnan(degreeOfFreedoms.upperBound))
+    {
+        if (mnMajorVersion > 1 || mnMinorVersion > 21)
+        {
+            tinyxml2::XMLElement* constraintElem = oXML.NewElement("Constraint");
+            constraintElem->SetAttribute("LowerBound", std::to_string(degreeOfFreedoms.lowerBound).c_str());
+            constraintElem->SetAttribute("UpperBound", std::to_string(degreeOfFreedoms.upperBound).c_str());
```

Comment:
Issue: These used to be outside the version check adding the attributes to the parent element.

Reply:
Nice catch. Fixed!